src/modules/flow/platform/platform.c

```diff
@@ -673,4 +673,17 @@ locale_apply_time_process(struct sol_flow_node *node, void *data, uint16_t port,
     return sol_platform_apply_locale(SOL_PLATFORM_LOCALE_TIME);
 }
 
+static int
+platform_target_process(struct sol_flow_node *node, void *data, uint16_t port,
+    uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    int r;
+    const char *target;
+
+    r = sol_flow_packet_get_string(packet, &target);
+    SOL_INT_CHECK(r, < 0, r);
+
+    return sol_platform_set_target(target);
```

Comment:
i wonder if we should just return the error like you did or send an error packet. I think that set-target may fail if there is an ongoing request, in this case it would be better to explicitly send an error packet saying so.

the return-error is more like the case of packet_get_string() failing, something that is not up to the user to handle